### DISCUSSION TITLE
Is there an option to swith off property type convertions

### DISCUSSION THREAD

**Discussion by ichalyshev** [riok/mapperly#211] [Q&A]
You made a great job! 

Is there an option to not generate type convertions? F.e.:

```
public class FromType
{
	public string Id { get; set; }	
}

public class ToType
{
	public int? Id { get; set; }
}

[Mapper()]
public partial class TestMapper
{
	public partial ToType MapFromTo(FromType source);
}
```

generated code 

```
 public partial TypesLibrary.ToType MapFromTo(TypesLibrary.FromType source)
{
	var target = new TypesLibrary.ToType();
	target.Id = int.Parse(source.Id);
	return target;
}
```

mapperly generate conversion:

target.Id = int.Parse(source.Id);

it is great but sometimes all i need is plain assignment target.Id = source.Id with compilation error - "Cannot implicitly convert type 'string' to 'int?' Is there an option for this?

**Comment by latonz** [comment]
Thank you for your feature request.
This feature could be implemented via a flag to disable all conversions except for direct assignments or it could allow better customisation via an enum flag which conversions to allow (eg. implicit casts, explicit casts, parse methods, ..).
Currently this is not on our list of features we plan to implement in the foreseeable future.

**Reply by ichalyshev** [reply]
Hi. 

>>>customisation via an enum flag which conversions to allow (eg. implicit casts, explicit casts, parse methods, ..).

Good decision. It doesn't look complex i can try implement little part of logic that i need right now f.e.

[Flag]
enum ConvertValueStrategy
{
None = 0,

All = ~None
}

public sealed class MapperAttribute : Attribute
...
public ConvertValueStrategy DefaultConvertValueStrategy { get; set; } = ConvertValueStrategy.All;

Wich can be specified on every mapping method (DefaultConvertValueStrategy ), and than for every property mapping (ConvertValueStrategy).

I can do it in my fork or in some other branch. I'm still did not worked with github.

**Reply by latonz** [reply]
This feature is released in `2.6.0-next.3`.

To disable all conversions supported by Mapperly set `EnabledConversions` on the `MapperAttribute` to `None`:
```csharp
[Mapper(EnabledConversions = MappingConversionType.None)]
public partial class CarMapper
{
  ...
}
```

To disable a specific conversion type, set `EnabledConversions` to `All` excluding the conversion type to disable:
```csharp
// this disables conversions using the ToString() method:
[Mapper(EnabledConversions = MappingConversionType.All & ~MappingConversionType.ToStringMethod)]
public partial class CarMapper
{
  ...
}
```

**Reply by ichalyshev** [reply]
That's great! Thanks a lot! 

I find serious restriction in source generators technology. Problem is source generators do not chain (isolated one from the other), and I still not found the solution. I use WFP toolkit in my project and it utilize source generators to generate properties in ViewModels partial classes. Since source generators isolated one from the other mapperly don't see autogenerated properties.

May be you know is there any solution?

**Reply by latonz** [reply]
Sorry, I can't help you with that. I don't know a solution to this.

**Reply by ichalyshev** [reply]
Anyway thanks a lot!
